helm/charts/determined/templates/master-service.yaml

```diff
@@ -7,6 +7,9 @@ metadata:
     app: determined-master-{{ .Release.Name }}
     determined.ai/master-service: "true"
     release: {{ .Release.Name }}
+  {{- if (.Values.masterService).annotations }}
```

Comment:
Remove unnecessary parentheses here as well.

Reply:
Without these parens it will cause a nil ptr.

https://stackoverflow.com/a/68807258/11775692

Reply:
Oh, I'm sorry. Usually there's a default map `{}` in values.yaml for top level stuff, like `masterService: {}`, but this works in its absence. `with` also works.